app/src/org/commcare/android/view/EntityView.java

```diff
@@ -505,8 +539,12 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
             }
         }
 
-        // Re-calculate the view's measurements based on the percentage
-        // adjustments above
+        // Re-calculate the view's measurements based on the percentage adjustments above
         super.onMeasure(widthMeasureSpec, heightMeasureSpec);
+
+        onMeasureCalled = true;
+        if (imageViewsToRedraw.size() > 0) {
+            redrawImageLayoutsInQueue();

```

Comment:
No, it doesn't- it just doesn't make a material difference in code execution because if the queue is empty then nothing happens (essentially the same reason you noted that the if statement is unnecessary).